packages/cdk/resources/CognitoFunctions.ts

```diff
@@ -0,0 +1,237 @@
+
+import {Construct} from "constructs"
+
+import {LambdaFunction} from "./LambdaFunction"
+import {ITableV2} from "aws-cdk-lib/aws-dynamodb"
+import {
+  AccountRootPrincipal,
+  Effect,
+  IManagedPolicy,
+  IRole,
+  ManagedPolicy,
+  PolicyDocument,
+  PolicyStatement
+} from "aws-cdk-lib/aws-iam"
+import {Duration, RemovalPolicy, SecretValue} from "aws-cdk-lib"
+import {Key} from "aws-cdk-lib/aws-kms"
+import {Secret} from "aws-cdk-lib/aws-secretsmanager"
+import {NodejsFunction} from "aws-cdk-lib/aws-lambda-nodejs"
+
+export interface CognitoFunctionsProps {
+  readonly serviceName: string
+  readonly stackName: string
+  readonly primaryOidcTokenEndpoint: string
+  readonly primaryOidcUserInfoEndpoint: string
+  readonly primaryOidcjwksEndpoint: string
+  readonly primaryOidcClientId: string
+  readonly primaryOidcIssuer: string
+  readonly useMockOidc: boolean
+  readonly mockOidcTokenEndpoint?: string
+  readonly mockOidcUserInfoEndpoint?: string
+  readonly mockOidcjwksEndpoint?: string
+  readonly mockOidcClientId?: string
+  readonly mockOidcIssuer?: string
+  readonly tokenMappingTable: ITableV2
+  readonly tokenMappingTableWritePolicy: IManagedPolicy
+  readonly tokenMappingTableReadPolicy: IManagedPolicy
+  readonly useTokensMappingKmsKeyPolicy: IManagedPolicy
+  readonly primaryPoolIdentityProviderName: string
+  readonly mockPoolIdentityProviderName: string
+  readonly logRetentionInDays: number
+  readonly deploymentRole: IRole
+
+}
+
+/**
+ * Functions and resources that are needed for cognito
+ */
+export class CognitoFunctions extends Construct {
+  public readonly cognitoPolicies: Array<IManagedPolicy>
+  public readonly tokenLambda: NodejsFunction
+  public readonly mockTokenLambda: NodejsFunction
+  public readonly primaryJwtPrivateKey: Secret
+
+  public constructor(scope: Construct, id: string, props: CognitoFunctionsProps) {
+    super(scope, id)
+
+    // Resources
+
+    // kms key used to encrypt the secret that stores the JWT private key
+    const jwtKmsKey = new Key(this, "JwtKMSKey", {
+      removalPolicy: RemovalPolicy.DESTROY,
+      pendingWindow: Duration.days(7),
+      alias: `${props.stackName}-JwtKMSKeyKMSKey`,
+      description: `${props.stackName}-JwtKMSKeyKMSKey`,
+      enableKeyRotation: true,
+      policy: new PolicyDocument({
+        statements: [
+          new PolicyStatement({
+            sid: "Enable IAM User Permissions",
+            effect: Effect.ALLOW,
+            actions: [
+              "kms:*"
+            ],
+            principals: [
+              new AccountRootPrincipal
+            ],
+            resources: ["*"]
+          }),
+          new PolicyStatement({
+            effect: Effect.ALLOW,
+            principals: [props.deploymentRole],
+            actions: [
+              "kms:Encrypt",
+              "kms:GenerateDataKey*"
+            ],
+            resources:["*"]
```

Comment:
as above

Reply:
it is not meaningless - it is required. See resources section of https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-overview.html#key-policy-elements